server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -1117,19 +1109,22 @@ public List<String> readImmutableStringList() throws IOException {
      *
      * @return the list of strings
      * @throws IOException if an I/O exception occurs reading the list
+     * @deprecated use {@link #readStringCollectionAsList}.
      */
+    @Deprecated(forRemoval = true)
     public List<String> readStringList() throws IOException {
-        return readList(StreamInput::readString);
+        return readStringCollectionAsList();
     }
 
     /**
      * Reads an optional list. The list is expected to have been written using
      * {@link StreamOutput#writeOptionalCollection(Collection)}. If the returned list contains any entries it will be mutable.
      * If it is empty it might be immutable.
+     * @deprecated use {@link #readOptionalCollectionAsList}.
      */
+    @Deprecated(forRemoval = true)
```

Comment:
Do we really need to go through the "fun" of deprecating and then taking forever to remove this stuff? :) Why not just outright do away with this thing ?

Reply:
Just to avoid breaking serverless - I'll remove it ASAP